src/hooks/v1/V1ContractLoader.ts

```diff
@@ -47,11 +47,13 @@ export function useV1ContractLoader() {
   return contracts
 }
 
-const loadContract = (
+const loadContract = async (
```

Comment:
**issue**: we need to also refactor the `loadContracts` function https://github.com/jbx-protocol/juice-interface/pull/1213/files#diff-7d6dcf148ae7c1e6fa95ffb7b7368ab023c49294d6df19028ac0ebc08a11622fR29

Check the V2 contract loader for an example; i recently refactored it to be async.

Reply:
Are you talking about `src/hooks/v1/V1ContractLoader.ts`? It looks like it is already async:
```ts
export function useV1ContractLoader() {
  const [contracts, setContracts] = useState<V1Contracts>()

  const { signingProvider } = useContext(NetworkContext)

  useEffect(() => {
    async function loadContracts() {
      try {
...
```

Reply:
Yeah i think we should `await` until all the contracts are loaded, just to be safe.

Check this diff, should be copy-pastable: https://github.com/jbx-protocol/juice-interface/pull/1085/files#diff-bf47168917ef85b811be6c44b2cdecaba47629ea198f83ac7ab493b20b5f076bR42-R56

Reply:
ah i see now - can do

Reply:
have done 👍